rs/rosetta-api/icrc1/ledger/tests/tests.rs

```diff
@@ -1543,3 +1553,151 @@ mod verify_written_blocks {
         }
     }
 }
+
+mod incompatible_token_type_upgrade {
+    use super::*;
+    use assert_matches::assert_matches;
+    use ic_state_machine_tests::ErrorCode::CanisterCalledTrap;
+    use num_bigint::BigUint;
+
+    fn default_init_args() -> Vec<u8> {
+        Encode!(&LedgerArgument::Init(InitArgs {
+            minting_account: MINTER,
+            fee_collector_account: None,
+            initial_balances: vec![],
+            transfer_fee: FEE.into(),
+            token_name: TOKEN_NAME.to_string(),
+            decimals: Some(DECIMAL_PLACES),
+            token_symbol: TOKEN_SYMBOL.to_string(),
+            metadata: vec![],
+            archive_options: ArchiveOptions {
+                trigger_threshold: ARCHIVE_TRIGGER_THRESHOLD as usize,
+                num_blocks_to_archive: NUM_BLOCKS_TO_ARCHIVE as usize,
+                node_max_memory_size_bytes: None,
+                max_message_size_bytes: None,
+                controller_id: PrincipalId::new_user_test_id(100),
+                more_controller_ids: None,
+                cycles_for_archive_creation: None,
+                max_transactions_per_response: None,
+            },
+            max_memo_length: None,
+            feature_flags: Some(FeatureFlags { icrc2: false }),
+            maximum_number_of_accounts: None,
+            accounts_overflow_trim_quantity: None,
+        }))
+        .unwrap()
+    }
+
+    #[test]
+    fn should_successfully_upgrade_ledger_from_u64_to_u256_to_u64_wasm() {
+        let env = StateMachine::new();
+        let ledger_id = env
+            .install_canister(ledger_mainnet_u64_wasm(), default_init_args(), None)
+            .unwrap();
+        // Create a large balance that is still smaller than the max token supply of u64::MAX
+        transfer(&env, ledger_id, MINTER, account(1), u64::MAX - 100);
```

Comment:
It's not clear to me what's the role of `-100` here. Why not using as balance `u64::MAX`?

Reply:
You're right, the `-100` doesn't make sense here. Changed to `u64::MAX`, and also adapted the comment.